tests/large-lib-test.py

```diff
@@ -0,0 +1,136 @@
+#
+###############################################################
+
+
+Print=Utils.Print
+errorExit=Utils.errorExit
+
+args=TestHelper.parse_args({"--kill-sig","--kill-count","--keep-logs"
+                            ,"--dump-error-details","-v","--leave-running","--clean-run"
+                            })
+pnodes=1
+total_nodes=3 # first one is producer, and last two are speculative nodes
+debug=args.v
+killEosInstances=not args.leave_running
+dumpErrorDetails=args.dump_error_details
+keepLogs=args.keep_logs
+killAll=args.clean_run
+relaunchTimeout=10
+# Don't want to set too big, trying to reduce test time, but needs to be large enough for test to finish before
+# restart re-creates this many blocks.
+numBlocksToProduceBeforeRelaunch=80
+numBlocksToWaitBeforeChecking=20
+
+Utils.Debug=debug
+testSuccessful=False
+
+seed=1
+random.seed(seed) # Use a fixed seed for repeatability.
+cluster=Cluster(walletd=True)
+walletMgr=WalletMgr(True)
+cluster.setWalletMgr(walletMgr)
+
+def relaunchNode(node: Node, chainArg="", skipGenesis=True, relaunchAssertMessage="Fail to relaunch"):
+   isRelaunchSuccess=node.relaunch(chainArg=chainArg, timeout=relaunchTimeout, skipGenesis=skipGenesis, cachePopen=True)
+   time.sleep(1) # Give a second to replay or resync if needed
+   assert isRelaunchSuccess, relaunchAssertMessage
+   return isRelaunchSuccess
+
+try:
+    TestHelper.printSystemInfo("BEGIN")
+
+    cluster.killall(allInstances=killAll)
+    cluster.cleanup()
+    walletMgr.killall(allInstances=killAll)
+    walletMgr.cleanup()
+
+    # set the last two nodes as speculative
+    specificExtraNodeosArgs={}
+    specificExtraNodeosArgs[1]="--read-mode speculative "
+    specificExtraNodeosArgs[2]="--read-mode speculative "
+
+    Print("Stand up cluster")
+    if cluster.launch(
+            pnodes=pnodes,
+            totalNodes=total_nodes,
+            totalProducers=1,
+            useBiosBootFile=False,
+            topo="mesh",
+            specificExtraNodeosArgs=specificExtraNodeosArgs) is False:
+        errorExit("Failed to stand up eos cluster.")
+
+    producingNode=cluster.getNode(0)
+    speculativeNode1=cluster.getNode(1)
+    speculativeNode2=cluster.getNode(2)
+
+    Print ("Wait for Cluster stabilization")
+    if not cluster.waitOnClusterBlockNumSync(3):
+        errorExit("Cluster never stabilized")
+    Print ("Cluster stabilized")
+
+    Print("Wait for producing {} blocks".format(numBlocksToProduceBeforeRelaunch))
+    producingNode.waitForBlock(numBlocksToProduceBeforeRelaunch, blockType=BlockType.lib)
+
+    Print("Kill all node instances.")
+    for clusterNode in cluster.nodes:
+        clusterNode.kill(signal.SIGTERM)
+    cluster.biosNode.kill(signal.SIGTERM)
+    Print("All nodeos instances killed.")
+
+    # Remove both state and blocks such that no replay happens
+    Print("Remove producer node's state and blocks directories")
+    shutil.rmtree(Utils.getNodeDataDir(0,"state"))
```

Comment:
if you would do any changes on that PR I suggest we will move that part to `Node.py`.
This looks generic functionality. It could look like that:
`specificNode.cleanDataDir(cleanBlocks=True, cleanState=True)`
`cleanBlocks` and `cleanState` can be defaulted to True
Not a big deal but if any changes to be done I think it worth to move to framework.

Reply:
Moved